src/annotator/frame-observer.js

```diff
@@ -45,16 +38,11 @@ export default class FrameObserver {
   /**
    * @param {HTMLIFrameElement} frame
    */
-  _addFrame(frame) {
-    if (FrameUtil.isAccessible(frame)) {
-      FrameUtil.isDocumentReady(frame, () => {
-        const frameWindow = /** @type {Window} */ (frame.contentWindow);
-        frameWindow.addEventListener('unload', () => {
-          this._removeFrame(frame);
-        });
-        this._handledFrames.push(frame);
-        this._onFrameAdded(frame);
-      });
+  async _addFrame(frame) {
+    if (isAccessible(frame)) {
+      await onDocumentReady(frame);
+      this._handledFrames.add(frame);
+      this._onFrameAdded(frame);
```

Comment:
Adding a `console.warn` in the else block would be useful, as long as this method is only being called for frames which have the `enable-annotation` attribute, so as not to generate a lot of console spam on ad-heavy web pages that contain many iframes.

Reply:
👍🏼

Reply:
I decided to split the PR in two, it is in the following up https://github.com/hypothesis/client/pull/3686